From 453c30492f2ce20c4631db16fea6757672685df6 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 10 Sep 2024 00:10:11 +0800
Subject: [PATCH 1/3] fix(evdev-mt-touchpad-tap.c): enable tap-to-click by
 default

As we know, most touchpads these days come with no buttons *and*, more
importantly, decent palm rejection and tap zones/recognition. The original
patch that disabled this default behavior was built on assumptions from
2014 and may well be accurate, but it would seem a bit bizzare today to
see a click-only default.

KDE 6 has enabled this behaviour by default since a year ago.[^1] What's
more, the GNOME folks have been wondering why they still needed to click
their touchpads (some distributions enables tap-to-click by default but
some others doesn't). A quick search would find us many users confused by
this default behavior.

This is a partial revert of:

  commit 2219c12c3aa45b80f235e761e87c17fb9ec70eae
  Author: Peter Hutterer <peter.hutterer@who-t.net>
  Date:   Tue Feb 4 10:38:21 2014 +1000

    touchpad: hook up to the tapping configuration

    Now that we have run-time changes of the tap.enabled state move the check
    to the IDLE state only. Otherwise the tap machine may hang if tapping is
    disabled while a gesture is in progress.

    Two basic tests are added to check for the tap default setting - which is now
    "tap disabled by default", for two reasons:
    * if you don't know that tapping is a thing (or enabled by default), you get
      spurious button events that make the desktop feel buggy.
    * if you do know what tapping is and you want it, you usually know where to
      enable it, or at least you can search for it.

    Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

[^1]: https://invent.kde.org/plasma/plasma-desktop/-/issues/97

---

Patch submitted but NOT accepted by the upstream:

https://gitlab.freedesktop.org/libinput/libinput/-/merge_requests/1047

---

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 src/evdev-mt-touchpad-tap.c | 18 +-----------------
 1 file changed, 1 insertion(+), 17 deletions(-)

diff --git a/src/evdev-mt-touchpad-tap.c b/src/evdev-mt-touchpad-tap.c
index 3c76c3d6..7f912276 100644
--- a/src/evdev-mt-touchpad-tap.c
+++ b/src/evdev-mt-touchpad-tap.c
@@ -1420,23 +1420,7 @@ tp_tap_config_is_enabled(struct libinput_device *device)
 static enum libinput_config_tap_state
 tp_tap_default(struct evdev_device *evdev)
 {
-	/**
-	 * If we don't have a left button we must have tapping enabled by
-	 * default.
-	 */
-	if (!libevdev_has_event_code(evdev->evdev, EV_KEY, BTN_LEFT))
-		return LIBINPUT_CONFIG_TAP_ENABLED;
-
-	/**
-	 * Tapping is disabled by default for two reasons:
-	 * * if you don't know that tapping is a thing (or enabled by
-	 *   default), you get spurious mouse events that make the desktop
-	 *   feel buggy.
-	 * * if you do know what tapping is and you want it, you
-	 *   usually know where to enable it, or at least you can search for
-	 *   it.
-	 */
-	return LIBINPUT_CONFIG_TAP_DISABLED;
+	return LIBINPUT_CONFIG_TAP_ENABLED;
 }
 
 static enum libinput_config_tap_state
-- 
2.47.1

